File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    honor = []
    for s in score:
        if len(honor) < k:
            honor.append(s)
        else:
            honor.sort()
            if honor[0] < s:
                honor.pop(0)
                honor.append(s)

        honor.sort()
        answer.append(honor[0])
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.swift
import Foundation

func solution(_ k:Int, _ score:[Int]) -> [Int] {
    var honor = [Int]()
    var answer = [Int]()
    for s in score {
        honor.append(s)
        honor.sort(by: >)
        while honor.count > k {
            honor.popLast()
        }
        answer.append(honor.last ?? 0)
    }
    return answer
}